modules/scheduler/gke-cluster/main.tf

```diff
@@ -196,9 +238,12 @@ resource "google_container_node_pool" "system_node_pools" {
   provider = google-beta
   count    = var.system_node_pool_enabled ? 1 : 0
 
-  project = var.project_id
-  name    = var.system_node_pool_name
-  cluster = google_container_cluster.gke_cluster.self_link
+  project  = var.project_id
+  name     = var.system_node_pool_name
+  cluster  = var.is_gke_sandbox ? google_container_cluster.gke_cluster.name : google_container_cluster.gke_cluster.self_link
```

Comment:
Can we create explicit variable called `use_gke_self_link` (and then set in blueprint). Again my intent not exposing sandbox in providers ?

Reply:
Introducing `cluster_reference_type` which takes the values: `{SELF_LINK, NAME}` with `SELF_LINK` being the default